### ISSUE TITLE
nuget.org website keeps asking to sign in every day

### ISSUE THREAD

**Issue by sokolovsa** [NuGet/NuGetGallery#8534]
### Describe the bug
Nuget.org website asks to sign in at least once every day.
I use Firefox v87

### To Reproduce
1. Open nuget.org in browser and sign in with microsoft account
2. Proceed to interact with the website and make sure you're still signed in.
3. Next day, open nuget.org again and notice that you're not signed in
4. Click "sign in",  enter your email
5. Notice, that it doesn't even ask for a password 
6. you're signed in again.

### Expected behavior
nuget.org does not force me to enter my email every day and stays in signed in state.

**Comment by joelverhagen** [comment]
@sokolovsa, as far as I know, this is expected. We have a short session duration configured so this is why you're seeing the sign in flow so frequently. I agree, it can be tedious!

@shishirx34, do you remember the reasoning for this short session duration or if it is required?

**Comment by shishirx34** [comment]
@sokolovsa - do you have multiple Microsoft accounts? Is this a work account(AAD) or a personal account? For step4. do you need to "select" your email or actually need to enter your email?

@joelverhagen  - I believe our auth session cookie has a 6 hour expiry. After which we need to re-auth with azure identity for fresh token, for security reasons. The "make sure you're still signed in" behavior is that of the azure identity authentication and not something we control. We'll need a repro to investigate if you need to actually enter an email instead of "selecting" the already authenticated account. 

Ideally if there is only one MSA account linked to a browser session we'd want to pick just that. However, we've got lots of reports of users accidentally using incorrect MSA account and therefore ended up creating a new NuGet account unintentionally. That is why, by design, we have an option to "Select" an account at which point a user can choose to sign in with a different MSA account that is linked to NuGet.

**Comment by augustoproiete** [comment]
@shishirx34 In case it helps: I use a dedicated browser for all things Microsoft (e.g. NuGet, Microsoft Learning, Channel9, Yammer, etc.) with a single Microsoft account (personal, not work/school) and I'm asked to log in to NuGet at least a couple of times a day - which seems roughly to match the 6 hours of the cookie expiration you mentioned.

**Comment by shishirx34** [comment]
@augustoproiete - when you say "asked to log in" that is to click "Sign in with Microsoft" on NuGet login page and then you need to "Select" your MSA account, correct? If yes, that is the designed behavior. If you need to "re-enter" your MSA email and password, that is not, please let us know if such is the case. 

Also unless I am mistaken, the 6 hour expiry is a "Sliding window" which means if you are active within that duration you shouldn't be logged out until 6 hours after you become inactive

**Comment by augustoproiete** [comment]
@shishirx34 I'm usually outside of the 6 hour sliding window then. This is what I mean:

- I log into NuGet at 8am, and go to [Manage Packages](https://www.nuget.org/account/Packages) to check something
- I leave the browser open for the rest of the day, but do not interact with the website
- At 6pm on the same day, I still have the same page open on the same tab, it looks like I'm logged in and all the menus are still there

![image](https://user-images.githubusercontent.com/177608/115800352-d0f1a100-a3b0-11eb-9e93-cf87540b7e93.png)

- I click on [Manage Packages](https://www.nuget.org/account/Packages) or any other operation that would require authentication, such as [Upload Package](https://www.nuget.org/packages/manage/upload), and then I'm asked to sign in again.

---

It would be great if the expiration could be longer (e.g. 30 days) on a sliding window.

**Comment by shishirx34** [comment]
Yeah we do not have a "auto sign-out" and hence if you keep the page inactive it seems like you are still logged in although the session is not valid. The expiration window was long before, however with some security requirements it was reduced down to 6 hours. Please feel free to open a feature ask in a new issue for extending the session activity timespan with your reasons, that way other community members can also pitch in. If there is a way to do this securely we can prioritize the investigation.

**Comment by augustoproiete** [comment]
> Please feel free to open a feature ask in a new issue for extending the session activity timespan

@shishirx34 Happy to do that if helps - but this seems to be pretty much what @sokolovsa is asking for in this issue here - no?
 
I threw in the 30-day sliding window as a suggestion, but overall I think @sokolovsa's goal is the same: Not have to log in every day as frequently when coming back to nuget.org

**Comment by zhhyu** [comment]
For the compliance requirement, we may need to provide the checkbox for customers to select if a longer/permanent duration of the authentication session cookie is needed.

**Comment by shishirx34** [comment]
@zhhyu - those were my thoughts, perhaps a setting to allow for longer session duration. But it needs a security review.

**Comment by skofman1** [comment]
The 6 hours window was chosen for security reasons. As far as I recall, if you interact with the website during the window, it will extend the "logged-in" window by another 6 hours. 
We will not extend the 6 hours window for security reasons.

**Comment by sokolovsa** [comment]
> @sokolovsa - do you have multiple Microsoft accounts? Is this a work account(AAD) or a personal account? For step4. do you need to "select" your email or actually need to enter your email?

I have only one Microsoft account.
I need to explicitly enter my email every day. First, I press "Sign In" link on top right of the nuget.org web page, then I press "Sign in  with Microsoft" button, and then I enter my email. There is no option to select email that was entered before. 
As I remember, this behaviour changed several months ago. Before, it was possible to just "select" my email, but I guess after some nuget.org update this option disappeared and now I enter my email at least once every day.

**Comment by sokolovsa** [comment]
> 
> 
> > Please feel free to open a feature ask in a new issue for extending the session activity timespan
> 
> @shishirx34 Happy to do that if helps - but this seems to be pretty much what @sokolovsa is asking for in this issue here - no?
> 
> I threw in the 30-day sliding window as a suggestion, but overall I think @sokolovsa's goal is the same: Not have to log in every day as frequently when coming back to nuget.org

Yes, I agree. 
I understand the security reasons but 6 hours window looks pretty short especially when compared to other websites with comparable security considerations, like gmail or github.
It would be great if the sliding window was a bit longer, at least 3-4 days so that it carries over through the weekend.
